diff -aur a/tools/shared.py b/tools/shared.py
--- a/tools/shared.py 2025-10-14 09:16:13.210070677 +0000
+++ b/tools/shared.py 2025-10-14 09:15:07.106178680 +0000
@@ -5,7 +5,7 @@
 
 from __future__ import print_function
 
-from distutils.spawn import find_executable
+from shutil import which
 from subprocess import PIPE, STDOUT
 import atexit
 import base64
@@ -212,10 +212,10 @@
   config_file = '\n'.join(config_file)
   # autodetect some default paths
   config_file = config_file.replace('\'{{{ EMSCRIPTEN_ROOT }}}\'', repr(__rootpath__))
-  llvm_root = os.path.dirname(find_executable('llvm-dis') or '/usr/bin/llvm-dis')
+  llvm_root = os.path.dirname(which('llvm-dis') or '/usr/bin/llvm-dis')
   config_file = config_file.replace('\'{{{ LLVM_ROOT }}}\'', repr(llvm_root))
 
-  node = find_executable('nodejs') or find_executable('node') or 'node'
+  node = which('nodejs') or which('node') or 'node'
   config_file = config_file.replace('\'{{{ NODE }}}\'', repr(node))
 
   abspath = os.path.abspath(os.path.expanduser(path))
